hyperspy/signal.py

```diff
@@ -1712,6 +1712,88 @@ def data(self, value):
 
+    @property
+    def real(self):
+        """Get the real part of the data. Returns an :class:`~hyperspy.signals.Signal2D`."""
+        sig = self._deepcopy_with_new_data(np.real(self.data))
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'real({})'.format(title)
+        return sig
+
+    @property
+    def imag(self):
+        """Get imaginary part of the data. Returns an :class:`~hyperspy.signals.Signal2D`."""
+        sig = self._deepcopy_with_new_data(np.imag(self.data))
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'imag({})'.format(title)
+        return sig
+
+    def angle(self, deg=False):
+        """Return the angle of the (complex) data of the signal. Is zero if data is real.Calculate
+        the complex anglet the phase of the data.
+
+
+        Parameters
+        ----------
+        deg : bool, optional
+            Return angle in degrees if True, radians if False (default).
+
+        Returns
+        -------
+        angle : :class:`~hyperspy.signals.Signal2D`
+            The counterclockwise angle from the positive real axis on the complex plane,
+            with dtype as numpy.float64.
+
+        """
+        sig = self._deepcopy_with_new_data(np.angle(self.data, deg))
+        sig.set_signal_type('')
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'angle({})'.format(title)
+        return sig
+
+    def unwrapped_phase(self, wrap_around=False, seed=None, show_progressbar=None):
+        """Return the unwrapped phase as an :class:`~hyperspy.signals.Signal2D`.
+
+        Parameters
+        ----------
+        wrap_around : bool or sequence of bool, optional
+            When an element of the sequence is  `True`, the unwrapping process
+            will regard the edges along the corresponding axis of the image to be
+            connected and use this connectivity to guide the phase unwrapping
+            process. If only a single boolean is given, it will apply to all axes.
+            Wrap around is not supported for 1D arrays.
+        seed : int, optional
+            Unwrapping 2D or 3D images uses random initialization. This sets the
+            seed of the PRNG to achieve deterministic behavior.
+        show_progressbar : None or bool
+            If True, display a progress bar. If None the default is set in
+            `preferences`.
+
+        Returns
+        -------
+        phase_image: :class:`~hyperspy._signals.Signal2D`
+            Unwrapped phase.
+
+        Notes
+        -----
+        Uses the :func:`~skimage.restoration.unwrap_phase` function from `skimage`.
```

Comment:
Maybe adding a reference to the original article Miguel Arevallilo Herraez, David R. Burton, Michael J. Lalor, and Munther A. Gdeisat, “Fast two-dimensional phase-unwrapping algorithm based on sorting by reliability following a noncontinuous path”, Journal Applied Optics, Vol. 41, No. 35, pp. 7437, 2002 ?